### PR TITLE
feat: Add query-tables method

### PR DESCRIPTION
**Pull request by mure**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nisystemlink-clients-python/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

- Adds method to query tables with LINQ filter

### Why should this Pull Request be merged?

- Queries are useful 😄 

### What testing has been done?

- Added integration test that exercises all parameters

### CODE REVIEW

File: nisystemlink/clients/core/_uplink/_base_client.py [hunk 1]
```diff
@@ -41,7 +42,7 @@ def _handle_http_status(response: Response) -> Optional[Response]:
 @dumps.to_json(JsonModel)
 def _deserialize_model(model_cls: Type[JsonModel], model_instance: JsonModel) -> Dict:
     """Turns a :class:`.JsonModel` instance into a dictionary for serialization."""
-    return model_instance.dict(by_alias=True, exclude_unset=True)
+    return json.loads(model_instance.json(by_alias=True, exclude_unset=True))
```
Comment (mure) [inline nisystemlink/clients/core/_uplink/_base_client.py:45]: Working around this issue: https://github.com/pydantic/pydantic/issues/1409
tl;dr: calling `dict()` on a model doesn't serialize datetimes like `json()` does, but the converter is expecting a dictionary.

File: nisystemlink/clients/dataframe/models/_query_tables_request.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    Substitutions for the query expression are indicated by non-negative
+    integers that are prefixed with the ``@`` symbol. Each substitution in the given
+    expression will be replaced by the element at the corresponding index
+    (zero-based) in this list. For example, ``@0`` in the filter expression will be
+    replaced with the element at the zeroth index of the substitutions list.
+    """
+
+    reference_time: Optional[datetime] = None
+    """The date and time to use as the reference point for `RelativeTime` filters,
+    including time zone information. Defaults to the current time in UTC."""
```
Comment (spanglerco) [inline nisystemlink/clients/dataframe/models/_query_tables_request.py:60]: We may want to clarify that the default is server-side. Maybe something like:

```suggestion
    including time zone information. Defaults to the time on the server in UTC."""
```

File: nisystemlink/clients/dataframe/models/_query_tables_request.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+from pydantic import StrictBool, StrictInt
+
+from ._order_by import OrderBy
+
+
+class QueryTablesRequest(WithPaging):
+    """Request parameters for querying tables."""
+
+    filter: str
+    """The table query filter in Dynamic LINQ format.
```
Comment (spanglerco) [inline nisystemlink/clients/dataframe/models/_query_tables_request.py:14]: Is there a way to insert a link in the generated documentation? It may be useful to link to the [Dynamic Linq](https://github.com/ni/systemlink-OpenAPI-documents/wiki/Dynamic-Linq-Query-Language) documentation like we do in the swagger UI. In particular, it gives the format for DateTime and TimeSpan filter values and available operators/methods.

File: nisystemlink/clients/dataframe/models/_query_tables_request.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      between now and the start of the current day
+    * ``RelativeTime.CurrentWeek``: TimeSpan representing the elapsed time
+      between now and the start of the current week
+    * ``RelativeTime.CurrentMonth``: TimeSpan representing the elapsed time
+      between now and the start of the current month
+    * ``RelativeTime.CurrentYear``: TimeSpan representing the elapsed time
+      between now and the start of the current year
+    """
+
+    substitutions: Optional[List[Union[StrictInt, StrictBool, str, None]]] = None
```
Comment (spanglerco) [inline nisystemlink/clients/dataframe/models/_query_tables_request.py:48]: It could be interesting to explore having the API accept datetime (and maybe something representing a TimeSpan) values and serializing them to strings in the correct format.

**Review by spanglerco — CHANGES_REQUESTED** [review]
It could be nice to have tests for integer and boolean filter substitutions too. Such as `supportsAppend == @0 and rowCount < @1`

**Review by cameronwaterman — APPROVED** [review]

File: nisystemlink/clients/dataframe/models/_query_tables_request.py (hunk 3 above)
Comment (mure) [reply nisystemlink/clients/dataframe/models/_query_tables_request.py]: Should be linking out to that page now

File: nisystemlink/clients/dataframe/models/_query_tables_request.py (hunk 4 above)
Comment (mure) [reply nisystemlink/clients/dataframe/models/_query_tables_request.py:50]: So client-side substitutions for Python objects? That sounds useful - I'll keep it in mind.

**Comment by mure** [comment]
> It could be nice to have tests for integer and boolean filter substitutions too. Such as `supportsAppend == @0 and rowCount < @1`

Added more filters to the test

File: nisystemlink/clients/dataframe/models/_query_tables_request.py (hunk 4 above)
Comment (spanglerco) [reply nisystemlink/clients/dataframe/models/_query_tables_request.py:50]: Yep. Could be worth adding a task/story. Similar to how we want to support a helper for reading and writing row data as native types instead of only strings.

**Review by spanglerco — APPROVED** [review]
